### ISSUE TITLE
Firefox Add-On UNAVAILABLE?

### ISSUE THREAD

**Issue by rykel** [miscavage/De-Mainstream-YouTube-Extension#45]
Hi, is this extension no longer available for Firefox?

**Comment by HauptmannTod** [comment]
Hello. Your poor users need help. This is no longer available for Firefox. What tools and how to compile it or where to download alternatively if you have Windows.

**Comment by DarkAlchy** [comment]
This is pretty much a dead add-on.

**Comment by srid** [comment]
Mozilla, which is behind Firefox, has a [vested interest](https://www.bing.com/search?q=mozilla+site%3Areclaimthenet.org) in blocking addons like this. You'd be better off using a Chrome-based browser like [Brave](https://reclaimthenet.org/private-web-browsers/)

**Comment by DarkAlchy** [comment]
I realize that, but I just don't like Chromium based browsers.  btw, Mozilla has been turning evil for a decade now, but became very obvious when Gab's addon for commenting on sites without comments turned on, or at all, came out.

**Comment by srid** [comment]
Try Vivaldi then.

**Comment by DarkAlchy** [comment]
I am using Brave this second.  I am impressed at how well it imported all of my Firefox bookmarks but there is something causing a bit of lag at times (might be grabbing all but one addon I had on Firefox into Brave).

In your opinion Brave or Vivaldi?  Brave has come a long way in the last 4-5 years since I tried it and Vivaldi (back then) was the worst of them all, but it may have changed (as Brave has) since then.

**Comment by DarkAlchy** [comment]
Yes, as I was in the settings I saw that IPFS and had no idea wth that was, but thanks to your link on it I now know.  I like it.

What I wonder is am I doubling up when I don't need to?  Brave has its ad blocking then I throw on umatrix, and ublock, and I wonder if that may interfere or just be redundant?  Happen to know?

**Comment by srid** [comment]
You don't need to install any adblocking extension, because Brave has a _builtin_ adblocker. I think it is called "Trackers & ads blocking" in settings.

**Comment by DarkAlchy** [comment]
Well, umatrix and ublock are not just ad blockers but I did notice Brave uses the ad's portion of ublock lists but they are not checked because I don't live in those countries.  Make me wonder if Ublock is redudant?  I do like how I can remove elements with the eye dropper using it though (not something built into Brave for some odd reason).

I did download this addon though ad boy does it ever work.  Works so well it slows the whole page down (lol).

**Comment by srid** [comment]
You can block elements via selector: https://support.brave.com/hc/en-us/articles/360018039072-How-do-I-block-additional-page-elements-

**Comment by DarkAlchy** [comment]
Then there is one last thing and I am done with Mozilla (been trying to get away from them for four years as I saw the writing on the wall with them many years ago) and that is the lack of the scroll bar.  I scrolled through my bookmarks and it took me over 30 seconds to do using the down arrow.  Bad, really bad.  I went looking, and people have been asking for this for 4 years now and many have said this is the deal breaker for them (as it is for me).  I am not sure why the Brave devs haven't added this.  When something isn't added that many have asked for over 3-5 years I must wonder why?  Has to be a reason, but it does hurt for those of use who research and need that scroll bar as time is money.

**Comment by srid** [comment]
I don't know, you can try asking them in their community forums.

I personally don't use browser bookmarks feature for anything more than 10 links. Usually I store links _in their context_ (which also allows multiple ways to categorize), on my notebooks ([Obsidian](https://obsidian.md/) is good for managing that).

**Comment by Ph0rk0z** [comment]
Screw brave and screw brave shills. The browser is "open source" but I've never seen an independent build. Plus they have some program that replaces your ads with theirs which they pay for with their own crypto that's fairly worthless.

I've seen this browser shilled everywhere and it's yet-another-chrome-fork that makes you use THEIR ad blocker and their tools. Sometimes built-ins aren't good. You know how MS bundled internet explorer or how google tries to bundle chrome with things.


But the main point to all this is that none of this solves our problem of hiding mainstream youtube channels. When this extensions worked well my youtube pages were filled with videos I actually wanted to watch and not the flotsam that youtube recommends. Can't have that when you're pushing your opinions on people. Poof... it's gone.

**Comment by DarkAlchy** [comment]
Well, I went to Brave, but not happy (not as happy as I was in FF), but Mozilla is as bad as Google.  Nothing else out there, Vivaldi sucked the first time I tried it a few years back, and it still sucks.

The man he cometh to take it all away. :(  Play by their rules giving your blood, sweat, tears, and MONEY or you don't get to play is the modern Internet way.  I hate the modern way where big corps are sucking us dry while making us all feel good about it and asking for more of the same, or even worse.

**Comment by srid** [comment]
Brave, the company, gets a lot of things right. Mainly, the prioritization of decentralization tech and being censorship-resistant from the get-go. You can't say the same for other browsers like Firefox ([Mozilla is now against these principles](https://reclaimthenet.org/?s=mozilla)). 

They are also working on a search engine which addresses non-transparent censorship of Google and the like: https://brave.com/search

And the browser is [open source](https://github.com/brave/brave-browser), so nothing prevents anyone from creating an 'independent build' stripped of crypto features. Personally, I'm not anti-crypto, so I appreciate its built-in features which allows me to use blockchain based [alternative social media](https://satellite.earth/@srid) without fuss.

**Comment by DarkAlchy** [comment]
I wonder why no party has forked it?  Maybe the masses feel it doesn't need to be forked?  I know this if you read the devs are pretty assholey akin to the style of Microsoft, and Mozilla.  At least that is what I gleam from looking over the years of wants, needs, and bug reports.

**Comment by srid** [comment]
> I personally don't use browser bookmarks feature for anything more than 10 links. Usually I store links _in their context_ (which also allows multiple ways to categorize), on my notebooks ([Obsidian](https://obsidian.md/) is good for managing that).

Example: https://themotte.zettel.page/Users/sridqc

**Comment by DarkAlchy** [comment]
I have over 700 bookmarks dating back to 2004 (I used to run a check program and delete dead links but not for eons) as I kept them from Netscape, to Firefox to now Brave.

**Comment by desbest** [comment]
For people with lots of internet bookmarks like me I use [Diigo](http://diigo.com) to store them.

**Comment by BrandonKMLee** [comment]
@srid Brave is too awesome, maybe Dissenter? But also why would FF have a vested interest against an Open internet?

**Comment by DarkAlchy** [comment]
> @srid Brave is too awesome, maybe Dissenter? But also why would FF have a vested interest against an Open internet?

Because Mozilla went rogue as there is far too much to gain.  #1 They are leftist.  #2 Money in data collection.  #3 People making back room deals with them.  The list is very long as to why.

Mozilla has a blog post about them (was a few years ago I think) no longer for an open Internet.  The writing was on the wall the moment they went after Dissenter to shut it down, and Mozilla has turned darker and darker.  Mozilla is no one's friend in 2021 so treat them like an enemy combatant at best.

**Comment by triazatriborinane** [comment]
Please host a .xpi file so Firefox users can use this.

**Comment by desbest** [comment]
I've got a spare copy of the .xpi file on my laptop so here's a download link.

https://www.mediafire.com/file/gsxfc2809gvtkww/demainstream+youtube.xpi/file

https://mega.nz/file/wIM3hICZ#hK5JQnpnyH3KpTrTrfT0Wxt2vQWzSSSM8PbU9tXLGvE

https://app.box.com/s/5idfc1jz77ob4haexthcs40svndzpj7m

**Comment by githubuser181226** [comment]
> I've got a spare copy of the .xpi file on my laptop so here's a download link.
> 
> [Mediafire](https://www.mediafire.com/file/gsxfc2809gvtkww/demainstream+youtube.xpi/file)
> 
> [Mega](https://mega.nz/file/wIM3hICZ#hK5JQnpnyH3KpTrTrfT0Wxt2vQWzSSSM8PbU9tXLGvE)
> 
> [Box.com](https://app.box.com/s/5idfc1jz77ob4haexthcs40svndzpj7m)

DUDE! MANY THANKS!
God bless you!!!

**Comment by cake1001** [comment]
> I've got a spare copy of the .xpi file on my laptop so here's a download link. And it's the latest version 1.0.5
> 
> [Mediafire](https://www.mediafire.com/file/gsxfc2809gvtkww/demainstream+youtube.xpi/file)
> 
> [Mega](https://mega.nz/file/wIM3hICZ#hK5JQnpnyH3KpTrTrfT0Wxt2vQWzSSSM8PbU9tXLGvE)
> 
> [Box.com](https://app.box.com/s/5idfc1jz77ob4haexthcs40svndzpj7m)

Cheers.
One question. 
Which folder should I place this file in?

**Comment by desbest** [comment]
You don't need to put it in a folder. Just open it in firefox. 
If double clicking the file doesn't work use Ctrl + O or Alt ---> File ---> Open
